source/AI.cpp

```diff
@@ -829,7 +834,13 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 				else
 					parent.reset();
 
-				it->SetParent(parent);
+				// Player-owned carriables should defer to player carrier if
+				// selected parent can't carry it.
+				if(it->IsYours() && parent && parent->GetParent() && !parent->CanCarry(*it))
+					parent = parent->GetParent();
```

Comment:
Do we need this? If the parent can't carry then it will trigger the big if statement above (twice a second) which would find a suitable carrier.

Reply:
@quyykk I added a comment so that its purpose is more clear.